docs/core/testing/unit-testing-best-practices.md

```diff
@@ -0,0 +1,331 @@
+# Unit Testing Best Practices
+
```

Comment:
We need the addition of the metadata block before the H1 (on line 1). It would be the following (season title and description to taste):

```yml
title: Recommendations for unit test libraries
description: Learn practices for building unit testing libraries that drive code quality and resilience
author: jpreese
ms.author: wiwagn
ms.date: 07/28/2018
```